zio-notion-core/src/test/scala/zio/notion/dsl/ColumnSpec.scala

```diff
@@ -18,76 +18,273 @@ object ColumnSpec extends ZIOSpecDefault {
-        val filter = $"$fakeName".asNumber == 2
-        assertTrue(filter == Number(fakeName, NumberPropertyFilter.Equals(2)))
+        val filter: Filter =
+          $"$fakeName".asNumber == 2 and
+            $"$fakeName".asNumber < 3 and
+            $"$fakeName".asNumber != 1 and
+            $"$fakeName".asNumber > 1 and
+            $"$fakeName".asNumber <= 2 and
+            $"$fakeName".asNumber >= 2 and
+            $"$fakeName".asNumber.isNotEmpty
+
+        assertTrue(
+          filter ==
+            Number(fakeName, NumberPropertyFilter.Equals(2))
+              .and(Number(fakeName, NumberPropertyFilter.LessThan(3)))
+              .and(Number(fakeName, NumberPropertyFilter.DoesNotEqual(1)))
+              .and(Number(fakeName, NumberPropertyFilter.GreaterThan(1)))
+              .and(Number(fakeName, NumberPropertyFilter.LessThanOrEqualTo(2)))
+              .and(Number(fakeName, NumberPropertyFilter.GreaterThanOrEqualTo(2)))
+              .and(Number(fakeName, PropertyFilter.IsNotEmpty(true)))
+        )
       },
       test("creates a title filter") {
-        val filter = $"$fakeName".asTitle startsWith fakeName
-        assertTrue(filter == Title(fakeName, TextPropertyFilter.StartsWith(fakeName)))
+        val filter =
+          $"$fakeName".asTitle startsWith fakeName and
+            ($"$fakeName".asTitle endsWith fakeName) and
+            $"$fakeName".asTitle.equals(fakeName) and
+            $"$fakeName".asTitle.doesNotEqual("something else") and
+            $"$fakeName".asTitle.contains(fakeName) and
+            $"$fakeName".asTitle.doesNotContain("xyz") and
+            $"$fakeName".asTitle.isNotEmpty
+
+        assertTrue(
+          filter == Title(fakeName, TextPropertyFilter.StartsWith(fakeName))
+            .and(Title(fakeName, TextPropertyFilter.EndsWith(fakeName)))
+            .and(Title(fakeName, PropertyFilter.Equals(fakeName)))
+            .and(Title(fakeName, PropertyFilter.DoesNotEqual("something else")))
+            .and(Title(fakeName, PropertyFilter.Contains(fakeName)))
+            .and(Title(fakeName, PropertyFilter.DoesNotContain("xyz")))
+            .and(Title(fakeName, PropertyFilter.IsNotEmpty(true)))
+        )
       },
       test("creates a rich-text filter") {
-        val filter = $"$fakeName".asRichText startsWith fakeName
-        assertTrue(filter == RichText(fakeName, TextPropertyFilter.StartsWith(fakeName)))
+        val filter =
+          $"$fakeName".asRichText startsWith fakeName and
+            $"$fakeName".asRichText.endsWith(fakeName) and
+            $"$fakeName".asRichText.equals(fakeName) and
+            $"$fakeName".asRichText.doesNotEqual("something else") and
+            $"$fakeName".asRichText.contains(fakeName) and
+            $"$fakeName".asRichText.doesNotContain("xyz") and
+            $"$fakeName".asRichText.isNotEmpty
+
+        assertTrue(
+          filter == RichText(fakeName, TextPropertyFilter.StartsWith(fakeName))
```

Comment:
Plutôt que de mettre les And, j'aurais mis la structure Filter avec la List directement pour bien montrer la corrélation de l'état DSL à l'état domaine